### PR TITLE
040 handle absence of metamask

### PR DESCRIPTION
**Pull request by dosjcq**
Handle lack of metamask

### CODE REVIEW

File: src/api/bridge.js [hunk 1]
```diff
@@ -1,50 +1,103 @@
...
 import L2BridgeAbi from '../abis/L2Bridge.json'
 
-const L1TokenAddress = Networks[4].tokenAddress
-const L1BridgeAddress = Networks[4].bridgeAddress
-const L2TokenAddress = Networks[97].tokenAddress
-const L2BridgeAddress = Networks[97].bridgeAddress
+// if (!window.ethereum) {
+//   console.log('error');
+//   throw new Error('Install metamask')
+// }
```
Comment (alex-lazarev) [inline src/api/bridge.js]: delete this

File: src/api/bridge.js [hunk 2]
```diff
@@ -1,50 +1,103 @@
...
-const L1Token = new ethers.Contract(L1TokenAddress, L1TokenAbi, signer)
-const L1Bridge = new ethers.Contract(L1BridgeAddress, L1BridgeAbi, signer)
-const L2Token = new ethers.Contract(L2TokenAddress, L2TokenAbi, signer)
-const L2Bridge = new ethers.Contract(L2BridgeAddress, L2BridgeAbi, signer)
+// export function HandleMetamaskError ({error, setError}) {
+//   if (!window.ethereum) {
+//     console.log('error');
+//     setError(true)
+//   }
+// }
```
Comment (alex-lazarev) [inline src/api/bridge.js]: delete this

File: src/api/bridge.js [hunk 3]
```diff
@@ -1,50 +1,103 @@
...
+
+export async function bridgeToken(chainId, tokenId, setPending, setTransactionStatus, setChange, change, setDisableButtons) {
 
   const Token = Contracts[chainId].Token
   const Bridge = Contracts[chainId].Bridge
   const BridgeAddress = Contracts[chainId].BridgeAddress
-
   const signerAddress = await signer.getAddress()
   const allowance = (await Token.getApproved(tokenId)).toString()
+  console.log(chainId);
```
Comment (alex-lazarev) [inline src/api/bridge.js]: delete this

File: src/api/fetchTokens.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ export function getTokensInfo() {
   return axios
     .get(`${host}/tokens/`)
     .then(response => {
+      console.log(response.data.results);
```
Comment (alex-lazarev) [inline src/api/fetchTokens.js]: delete this

File: src/api/statusHandler.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+export function getTransactionStatus() {
+//   return axios
+//     .get(`${host}/tokens/`)
+//     .then(response => {
+//       return response.data.results
+//     })
+//     .catch(error => {
+//       console.log(error)
+//     })
```
Comment (alex-lazarev) [inline src/api/statusHandler.js]: delete this

File: src/api/statusHandler.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    axios
+        .get(`${host}/tokens/`)
+        .then(response => {
+        return response.data.results
+        })
+        .catch(error => {
+        console.log(error)
+    })
+}
```
Comment (alex-lazarev) [inline src/api/statusHandler.js]: new line needed

File: src/components/ErrorBoundary/ErrorBoundary.jsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            <br />
+            {this.state.errorInfo.componentStack}
+          </details>
+        </div>
+      );
+    }
+    // Normally, just render children
+    return this.props.children;
+  }  
+}
```
Comment (alex-lazarev) [inline src/components/ErrorBoundary/ErrorBoundary.jsx]: new line needed

File: src/components/Gallery/Gallery.jsx [hunk 8]
```diff
@@ -48,12 +50,14 @@ function TokensList({ tokens }) {
...
 export function Gallery() {
   const [tokensList, setTokensList] = useState([])
+  const [change, setChange] = useState(true);
 
   useEffect(() => {
     getTokensInfo().then(res => {
       setTokensList(res)
     })
-  }, [])
+    console.log(change);
```
Comment (alex-lazarev) [inline src/components/Gallery/Gallery.jsx]: delete this

File: src/components/Web3Status/Web3Status.jsx [hunk 9]
```diff
@@ -39,6 +39,8 @@ function NetworkSwitcher() {
     [NetworksMenuHidden]: !menuIsVisible,
   })
 
+  console.log(supportedChainIds);
```
Comment (alex-lazarev) [inline src/components/Web3Status/Web3Status.jsx]: delete this

File: src/index.jsx [hunk 10]
```diff
@@ -6,11 +6,22 @@ import './scss/index.scss'
 import { App } from './App'
 import { getLibrary } from './utils/web3/getLibrary'
 
+import {ErrorBoundary} from './components/ErrorBoundary/ErrorBoundary'
+
+
+// if (!window.ethereum) {
+//   console.log('error');
+//   throw new Error('eror')
+// }
```
Comment (alex-lazarev) [inline src/index.jsx]: delete this

File: src/networks.json [hunk 11]
```diff
@@ -1,38 +1,36 @@
...
-        "symbol": "ETH",
+        "name": "Ganache eth",
+        "symbol": "Eth",
         "decimals": 18
       },
-      "rpcUrls": [
-        "https://rinkeby.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161"
-      ],
-      "blockExplorerUrls": ["https://rinkeby.etherscan.io"]
+      "rpcUrls": ["https://0.0.0.0:8545"],
```
Comment (alex-lazarev) [inline src/networks.json]: https://127.0.0.1:8545

File: src/networks.json [hunk 12]
```diff
@@ -1,38 +1,36 @@
...
+      "chainName": "Ganache bsc",
       "nativeCurrency": {
-        "name": "Binance Coin",
+        "name": "Ganache bsc",
         "symbol": "BNB",
         "decimals": 18
       },
-      "rpcUrls": ["https://data-seed-prebsc-1-s1.binance.org:8545/"],
-      "blockExplorerUrls": ["https://bscscan.com"]
+      "rpcUrls": ["https://0.0.0.0:8546"],
```
Comment (alex-lazarev) [inline src/networks.json]: https://127.0.0.1:8546

File: src/utils/web3/getLibrary.js [hunk 13]
```diff
@@ -1,6 +1,7 @@
 import { Web3Provider } from '@ethersproject/providers'
 
 export function getLibrary(provider) {
+  console.log(provider);
```
Comment (alex-lazarev) [inline src/utils/web3/getLibrary.js]: delete this

File: src/utils/web3/hooks.js [hunk 14]
```diff
@@ -35,11 +35,12 @@ export function useInactiveListener(suppress = false) {
 
   useEffect(() => {
     const { ethereum } = window
+
     if (ethereum && ethereum.on && !active && !error && !suppress) {
       const handleChainChanged = (chainId) => {
         console.log('chainChanged', chainId)
         activate(injected)
-      }
+      }
```
Comment (alex-lazarev) [inline src/utils/web3/hooks.js]: cancel this

File: src/api/bridge.js [hunk 15]
```diff
@@ -1,50 +1,103 @@
...
             Bridge.outboundTransfer(signerAddress, tokenId).then(tx => {
+              setTransactionStatus(`Outbound with hash: ${shortenAddress(tx.hash)}`);
               tx.wait().then(() => {
-                setPending(false)
+                setTransactionStatus(`Mined with hash: ${shortenAddress(tx.hash)}`);
+                setTimeout(function testTokens() {
+                  axios
+                    .get(`${host}/tokens/`)
+                    .then(response => {
+                      console.log(response.data.results[tokenId])
```
Comment (alex-lazarev) [inline src/api/bridge.js]: delete this

File: src/api/bridge.js [hunk 16]
```diff
@@ -1,50 +1,103 @@
...
+                      if (response.data.results[tokenId].chain_id !== chainId) {
+                        setPending(false)
+                        setChange(change => !change)
+                        setDisableButtons(false)
+                        console.log('succwsss');
```
Comment (alex-lazarev) [inline src/api/bridge.js]: delete this

File: src/api/bridge.js [hunk 17]
```diff
@@ -1,50 +1,103 @@
...
+                      } else {
+                        console.log('not succes');
```
Comment (alex-lazarev) [inline src/api/bridge.js]: delete this

**Comment by Onbridge-io** [comment]
already in main

**Pull request closed without merging**